### PR TITLE
[RSDK-1596] Add Stop Signal/Timeout fields to ProcessConfig

### PR DESCRIPTION
**Pull request by Otterverse**
This supports the two new optional fields in goutil's pexec.ProcessConfig

### CODE REVIEW

File: proto/viam/app/v1/robot.proto [hunk 1]
```diff
@@ -88,6 +88,8 @@ message ProcessConfig {
   string cwd = 4;
   bool one_shot = 5;
   bool log = 6;
+  int32 stop_signal = 7;
```
Comment (edaniels) [inline proto/viam/app/v1/robot.proto:91]: opt: makeit an enum

File: proto/viam/app/v1/robot.proto [hunk 2]
```diff
@@ -88,6 +88,8 @@ message ProcessConfig {
...
+  int64 stop_timeout = 8;
```
Comment (edaniels) [inline proto/viam/app/v1/robot.proto:92]: seems like you should use proto duration for this

**Review by edaniels — CHANGES_REQUESTED** [review]

File: proto/viam/app/v1/robot.proto (hunk 1 above)
Comment (Otterverse) [reply proto/viam/app/v1/robot.proto:91]: I'd rather not enum it at the proto level. On a few weird architectures (e.g. if we ever supported some MIPS or SPARC board for some reason), some of the signals are different, so an int would still let something pass the "true" signal without needing a proto change, and the enum could filter at the application level.

File: proto/viam/app/v1/robot.proto (hunk 2 above)
Comment (Otterverse) [reply proto/viam/app/v1/robot.proto]: Good call... I didn't know that was a thing (I knew about timestamp, but didn't realize specifically duration was available.)

File: proto/viam/app/v1/robot.proto (hunk 1 above)
Comment (edaniels) [reply proto/viam/app/v1/robot.proto:91]: good point.

**Review by edaniels — APPROVED** [review]

**Pull request merged**